src/api/python/z3/z3.py

```diff
@@ -4456,9 +4461,12 @@ def Select(a, i):
     True
     """
     if z3_debug():
-        _z3_assert(is_array(a), "First argument must be a Z3 array expression")
-    return a[i]
-
+        _z3_assert(is_array_sorted_expr(a), "First argument must be a Z3 array expression")
+    if is_array(a):
+        return a[i]
```

Comment:
why are there two cases? both should becomes mk_select. The bug is the test "is_array" instead of "is_array_sorted_expr".

Reply:
lambda is a `QuantifierRef`. `QuantifierRef` does not implement `__getitem__`. 

The code can be moved into `__getitem__` of `QuantifierRef`, but then it should check that the quantifier is a lambda and not something else. I think it is cleaner the way it is now.